src/azure-cli/azure/cli/command_modules/acr/artifact_streaming.py

```diff
@@ -0,0 +1,381 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import time
+from enum import Enum
+from knack.log import get_logger
+from knack.util import CLIError
+from ._docker_utils import (
+    request_data_from_registry,
+    get_access_credentials,
+    RepoAccessTokenPermission,
+    RegistryException
+)
+from .manifest import(
+    _get_referrers_path,
+    _obtain_referrers_from_registry
+)
+from .repository import get_image_digest
+
+logger = get_logger(__name__)
+
+
+class OperationActions(str, Enum):
+    CANCEL = 'cancel'
+
+
+class ArtifactStreamingStatus(str, Enum):
+    NotStarted = 'NotStarted'
+    Running = 'Running'
+    Succeeded = 'Succeeded'
+    Failed = 'Failed'
+    Canceled = 'Canceled'
+
+
+def _get_v1_artifact_streaming_path(repository):
+    return '/acr/v1/{}/_streaming'.format(repository)
+
+
+def _get_v1_artifact_streaming_image_path(repository, digest):
+    return '/acr/v1/{}/_streaming/{}'.format(repository, digest)
+
+
+def _get_v1_artifact_streaming_operation_path(repository, operation_id, operation_type=None):
+    if operation_type:
+        return '/acr/v1/{}/_operations/{}:{}'.format(repository, operation_id, operation_type)
+    return '/acr/v1/{}/_operations/{}'.format(repository, operation_id)
+
+
+def acr_artifact_streaming_create(cmd,
```

Comment:
Do we validate if the registry is premium?

Reply:
We do not. But we're going to surface the detail errors to provide the costumer with more accurate information.